packages/seacas/libraries/ioss/src/Ioss_Region.C

```diff
@@ -3055,4 +3042,55 @@ namespace Ioss {
       topologyObserver->reset_topology_modification();
     }
   }
+
+  std::string Region::get_change_set_name() const
+  {
+    return get_database()->get_change_set_name();
+  }
+
+  std::tuple<std::string, int, double> Region::locate_db_state(double targetTime) const
+  {
+    IfDatabaseExistsBehavior  ifDatabaseExists{Ioss::DB_OVERWRITE};
+    unsigned int              dbChangeCount{1};
+
+    DynamicTopologyFileControl fileControl(const_cast<Region*>(this), get_file_cyclic_count(),
+                                           ifDatabaseExists, dbChangeCount);
+
+    return fileControl.locate_db_state(targetTime);
+  }
+
+  std::tuple<std::string, int, double> Region::get_db_max_time() const
+  {
+    IOSS_FUNC_ENTER(m_);
+    auto db = get_database();
+    if (!db->is_input() && db->usage() != WRITE_RESULTS && db->usage() != WRITE_RESTART) {
+      return std::make_tuple(get_change_set_name(), currentState, stateTimes[0]);
+    }
+
+    IfDatabaseExistsBehavior  ifDatabaseExists{Ioss::DB_OVERWRITE};
+    unsigned int              dbChangeCount{1};
+
+    DynamicTopologyFileControl fileControl(const_cast<Region*>(this), get_file_cyclic_count(),
+                                           ifDatabaseExists, dbChangeCount);
+
+    return fileControl.get_db_max_time();
+  }
+
+  std::tuple<std::string, int, double> Region::get_db_min_time() const
+  {
+    IOSS_FUNC_ENTER(m_);
+    auto db = get_database();
+    if (!db->is_input() && db->usage() != WRITE_RESULTS && db->usage() != WRITE_RESTART) {
```

Comment:
I guess this is for heartbeat and history?

Reply:
I wonder if we can simplify that line for input only ... can an input file have usage of `WRITE_RESULTS` or `WRITE_RESTART` ?

Reply:
It is for non-input (i.e. output) and I think the only output tyeps are WRITE_{results|restart|history|heartbeat}
(packages/seacas/libraries/ioss/src/Ioss_DBUsage.h)

Reply:
So yeah, this function is meant for RESULTS and RESTART, hence the early return ... not sure if to limit it to output (WRITE_*)  though